### ISSUE TITLE
Cannot access a closed file

### ISSUE THREAD

**Issue by margaale** [Zaid-Ajaj/Nancy.Serilog#5]
Sending a very long request as body in json format results in the same error #2 did. Following your advice there i ignored the contentbody as a hack to avoid the error. Tell me how o what can i do to help you.

Regards

**Comment by Zaid-Ajaj** [comment]
Hello Marcelo, thanks for filing the issue! 

Do you get the same error for any request or just requests where the body JSON is long? I will check...

**Comment by Zaid-Ajaj** [comment]
The example application seems to work just fine *without* needing to ignore any propertry:
```cs
// Bootstrapper.cs -> Application
protected override void ApplicationStartup(TinyIoCContainer container, IPipelines pipelines)
{
    // default config
    pipelines.EnableSerilog();

    Log.Logger = new LoggerConfiguration()
        .WriteTo.Console()
        .WriteTo.Console(new JsonFormatter())
        .CreateLogger();
}
``` 
then in request:
```cs
Post("/echo", async args => 
{
    await Task.Delay(1000);
    using (var reader = new StreamReader(this.Request.Body))
    {
        var content = await reader.ReadToEndAsync();
        return Response.AsText(content, "text/plain", Encoding.UTF8);
    }
});
```
Now sending a request to `/echo` this is what I get back:
```
[21:24:16 INF] Request POST /echo
{"Timestamp":"2019-06-12T21:24:16.0160723+02:00","Level":"Information","MessageTemplate":"Request {Method} {Path}","Properties":{"Method":"POST","Path":"/echo","RequestId":"0dc05278-8e27-418b-b4a5-81d66a1080a4","QueryString":"","RequestHostName":"localhost","RequestContentLength":15,"RequestContentType":"text/plain","RequestBodyContent":"This is my data","UserIPAddress":"::1","RequestHeaders":{"Cache-Control":"no-cache","Connection":"keep-alive","Content-Type":"text/plain","Accept":"application/vnd.heroku+json; version=3","Accept-Encoding":"gzip, deflate","Authorization":"Bearer d79fc99f-3b4d-485d-b557-0ef664618fa0","Host":"localhost:8080","User-Agent":"PostmanRuntime/7.13.0","Content-Length":"15","Postman-Token":"9ba6feee-53df-41e0-b9d2-6cef8defd626"},"Query":{},"RequestCookies":{},"UserAgentDevice":"Other","UserAgentFamily":"Other","UserAgentOS":"Other"}}
[21:24:17 INF] Response [200] POST /echo took 1037 ms
{"Timestamp":"2019-06-12T21:24:17.0266095+02:00","Level":"Information","MessageTemplate":"Response [{StatusCode}] {Method} {Path} took {Duration} ms","Properties":{"StatusCode":200,"Method":"POST","Path":"/echo","Duration":1037,"RequestId":"0dc05278-8e27-418b-b4a5-81d66a1080a4","ResponseContentType":"text/plain; charset=utf-8","ReasonPhrase":"","ResolvedPath":"/echo","RequestedPath":"/echo","ResponseHeaders":{},"ResponseContent":"This is my data","ResponseContentLength":15,"RawResponseCookies":[],"ResponseCookies":{},"ResolvedRouteParameters":{}}}
```
Do you have an example where it isn't working?

**Comment by margaale** [comment]
Not much diferent from yours... But I'm sending a json body with 116KB of data.

**Comment by Zaid-Ajaj** [comment]
Strange indeed, I was able to reproduce the problem using a large body request content but now I have fixed it and published a new version 2.1.0 that will soon be available on nuget :smile: 

Can you please confirm that it works when it becomes available? Please re-open the issue if the problem persists

**Comment by margaale** [comment]
Thanks for your hard work, now it's working fine.
